File: Backend/Academy.Shared/DTO/DashboardDataModel.cs
namespace Academy.Shared.DTO
{
    public class DashboardDataModel
    {
        public List<InterviewPanelModel> PanelData { get; set; }
        public DashboardTilesDataModel DashboardTiles { get; set; }
        public List<ChartDataModel> CommunityChartDataModel { get; set; }
        public List<ChartDataModel> PanelTypeChartDataModel { get; set; }
        public List<InterviewScheduleData> InterviewScheduleData { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}

File: Frontend/Academy.FE/src/app/components/agk-panel-availability/model/dashboard-data.model.ts
import { InterviewScheduleData } from "../interview-schedule/heatmap-table/heatmap-table.model";
import { ChartDataModel } from "./chartdata.model";
import { DashboardTileModel } from "./dashboard-tile.model";
import { PanelGrid } from "./panel-grid.model";

export class DashboardDataModel
{
    dashboardTiles: DashboardTileModel | undefined;
    communityChartDataModel: ChartDataModel[] = [];
    panelTypeChartDataModel: ChartDataModel[] = [];
    panelists: InterviewScheduleData[] = [];
    startDate: Date | undefined;
    endDate: Date | undefined;


    constructor(obj?: any){
        Object.assign(this,obj);
    }
}
